### PR TITLE
feat: add display prop to Grid component for SSR support

### PR DESCRIPTION
**Pull request by kim-msft**
# Description

## Motivation & context

Related to: [https://github.com/microsoft/fast-dna/issues/2408](https://github.com/microsoft/fast-dna/issues/2408)

For server side rendering we want to ensure we use the same grid styles that the client would produce. To do this we want to be able to set the grid styles ourselves instead of calling canUseCssGrid() which defaults to false on the server.

This change adds a new prop "display" to the grid component which can be one of "grid" or "-ms-grid". When provided this prop will be used to determine the styles that are used.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-layouts-react/src/grid/grid.props.ts [hunk 1]
```diff
@@ -69,6 +74,12 @@ export interface GridHandledProps extends GridManagedClasses {
      * Use this property to support -ms-grid when Grid is a child of an element who's display property is set to -ms-grid
      */
     row?: number;
+
+    /**
+     * The CSS grid display, "grid" when true and "-ms-grid" when false
+     * Provide this prop when doing server side rendering
+     */
+    display?: GridDisplay;
```
Comment (chrisdholt) [inline packages/fast-layouts-react/src/grid/grid.props.ts]: I'm torn on the naming convention here a bit. `display` initially causes me to think this relates to whether the grid should be displayed or not - though I do understand we have the type system providing context. I do wonder if a modifier may possibly help here? Perhaps something like `displayType` or something which eludes a bit more to it being related to the display property?

Comment (kim-msft) [reply packages/fast-layouts-react/src/grid/grid.props.ts]: I can see how display is a bit vague, displayType sounds good! I'll make the change

File: packages/fast-layouts-react/src/grid/grid.props.ts [hunk 2]
```diff
@@ -30,6 +30,11 @@ export enum GridAlignment {
  */
 export type GridGutter = 0 | 2 | 3 | 6 | 12;
 
+/**
+ * The CSS grid display style
+ */
+export type GridDisplay = "grid" | "-ms-grid";
```
Comment (chrisdholt) [inline packages/fast-layouts-react/src/grid/grid.props.ts]: @kim-msft can you make sure this is being exported from the `index.ts` file?

Comment (kim-msft) [reply packages/fast-layouts-react/src/grid/grid.props.ts]: Updated - thank you!

File: packages/fast-layouts-react/src/grid/grid.props.ts [hunk 3]
```diff
@@ -69,6 +74,12 @@ export interface GridHandledProps extends GridManagedClasses {
      * Use this property to support -ms-grid when Grid is a child of an element who's display property is set to -ms-grid
      */
     row?: number;
+
+    /**
+     * The CSS grid display, "grid" when true and "-ms-grid" when false
+     * Provide this prop when doing server side rendering
+     */
+    displayType?: GridDisplay;
```
Comment (janechu) [inline packages/fast-layouts-react/src/grid/grid.props.ts]: This seems a little misleading, as what it is is a CSS property name for the grid property. My preference would be:
`cssGridPropertyName`.

Comment (chrisdholt) [reply packages/fast-layouts-react/src/grid/grid.props.ts]: I'm good with this.

Comment (kim-msft) [reply packages/fast-layouts-react/src/grid/grid.props.ts]: sounds good

File: packages/fast-layouts-react/src/grid/grid.no-css-grid.spec.tsx [hunk 4]
```diff
@@ -78,3 +78,85 @@ describe("Grid - without CSS grid support", (): void => {
         expect(rendered.props().style.msGridColumn).toBe(2);
     });
 });
+
+describe("Grid - without CSS grid support but `display` prop is `grid`", (): void => {
```
Comment (janechu) [inline packages/fast-layouts-react/src/grid/grid.no-css-grid.spec.tsx]: This should be updated to `cssGridPropertyName` (same below)

Comment (kim-msft) [reply packages/fast-layouts-react/src/grid/grid.no-css-grid.spec.tsx]: Updated, thanks!

**Comment by chrisdholt** [comment]
Merged - thanks @kim-msft!

**Pull request merged**